src/zope/app/server/tests/test_mkzopeinstance.py

```diff
@@ -18,42 +18,44 @@
 import sys
 import tempfile
 import unittest
-import zope.app.server
-
-from StringIO import StringIO
+from contextlib import contextmanager
 
 from zope.app.server import mkzopeinstance
 
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
 
-class TestBase(unittest.TestCase):
 
-    def setUp(self):
-        self.stdout = StringIO()
-        self.stderr = StringIO()
-        self.old_stdout = sys.stdout
-        self.old_stderr = sys.stderr
-        sys.stdout = self.stdout
```

Comment:
Thank you for limiting the scope of the replacement!

Reply:
The bit where this was hiding failure tracebacks from me was simply unacceptable!